### PR TITLE
Migrate pagination to vue3

### PR DESCRIPTION
**Pull request by tahmeedshah**
### General description
add a custom pagination component.
remove the external pagination

### Designs
_Link to the related design prototypes (if applicable)_

### Testing instructions
_Provide minimal instructions on how to test this PR._

* Apart from the added feature / bug fix, check overall performance, styling...

## Checklist before merging
* [ ]  Branch name / PR includes the related Jira ticket Id.
* [ ]  Tests to check core implementation / bug fix added.
* [ ]  All checks in Continuous Integration workflow pass.
* [ ]  Feature functionally tested by reviewer(s).
* [ ]  Code reviewed by reviewer(s).
* [ ]  Documentation updated (README, CHANGELOG...) (if required)

### CODE REVIEW

File: app/components/common/pagination.vue [hunk 1]
```diff
@@ -0,0 +1,111 @@
+<template>
+  <nav v-if="recordCount" aria-label="Pagination" class="pagination d-flex justify-content-center text-center">
```
Comment (blaisefonseca) [inline app/components/common/pagination.vue:2]: aria-label="Pagination" get from translation json

File: app/components/common/pagination.vue [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    </ul>
+  </nav>
+
+</template>
+
+<script setup>
+import { ref, computed, defineProps, defineEmits } from 'vue';
+
+const { recordCount, recordsPerPage, currentPage, firstLastButton, firstButtonText, lastButtonText, prevText, nextText } = defineProps({
```
Comment (blaisefonseca) [inline app/components/common/pagination.vue:41]: destructing props will lose reactivity

File: app/components/common/pagination.vue [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+});
+
+const emit = defineEmits(['changePage']);
+
+const internalCurrentPage = ref(props.currentPage);
+
+const setPage = (page) => {
+  internalCurrentPage.value = page;
+  emit('changePage', page);
+};
```
Comment (sbilodeau) [inline app/components/common/pagination.vue:62]: is there a reason to have internalCurrentPageif the value is pushed to the parent `v-model:current-page`?
why not make it computer with setter? make be it was not designed this way.
eg:
```suggestion
const internalCurrentPage = computed({
  get() { return props.currentPage; }
  set(page) { emit('update:changePage', page); }
});ref(props.currentPage);
```

**Review by sbilodeau — APPROVED** [review]

File: app/components/common/pagination.vue (hunk 3 above)
Comment (tahmeedshah) [reply app/components/common/pagination.vue]: use computed properties.

**Pull request merged**